lib/target.py

```diff
@@ -223,8 +311,8 @@ class Vendor:
 
 class Target:
     triple = None
-    sdk = OSType.MacOSX
-    arch = ArchType.x86_64
+    sdk = OSType.UnknownOS
+    arch = ArchType.UnknownArch
     executable_suffix = ""
     dynamic_library_prefix = "lib"
     dynamic_library_suffix = ".dylib"
```

Comment:
I think I added the appropriate logic to compensate below.  The only issue I have with setting default values is that it increased the time it took to identify the logic bug that "ArchType.from_string(comps[0])" wasn't actually assigning anything, it was essentially a no-op.  The default value made it look like everything was fine, just wrong.